ArchiSteamFarm/Bot.cs

```diff
@@ -554,6 +567,43 @@ internal sealed class Bot : IDisposable {
 			return result;
 		}
 
+		internal async Task<Dictionary<string, string>> GetKeysFromFile(string filePath) {
+			if (string.IsNullOrEmpty(filePath) || !File.Exists(filePath)) {
+				ArchiLogger.LogNullError(nameof(filePath));
+				return null;
+			}
+
+			try {
+				Dictionary<string, string> keys = new Dictionary<string, string>();
+
+				using (StreamReader reader = new StreamReader(filePath)) {
+					string line;
+
+					while ((line = await reader.ReadLineAsync().ConfigureAwait(false)) != null) {
+						if (line.Length == 0) {
+							continue;
+						}
+
+						string[] parsedArgs = line.Split(DefaultBackgroundKeysRedeemerSeparator, StringSplitOptions.RemoveEmptyEntries);
+						if (parsedArgs.Length < 4) {
+							ArchiLogger.LogGenericWarning(string.Format(Strings.ErrorIsInvalid, line));
+							continue;
+						}
+
+						string name = parsedArgs[0];
+						string key = parsedArgs[3];
```

Comment:
This won't work for more arguments than 4, you should instead use `parsedArgs[parsedArgs.Length - 1]`.